tests/lit-tests/loop-codegen-acc-width.ispc

```diff
@@ -0,0 +1,86 @@
+// Test ensures performant codegen for accumulations over for loops
+// See [GitHub Issue core/#2460](https://github.com/ispc/ispc/issues/2460)
+
+// RUN: %{ispc} -o %t.asm %s --emit-asm --nostdlib --target=avx512skx-x4 --arch=x86-64 -DEXPLICIT_VECTOR4=0
+// RUN: FileCheck %s --input-file=%t.asm --check-prefix=CHECK_VECTOR0
+// RUN: %{ispc} -o %t.asm %s --emit-asm --nostdlib --target=avx512skx-x4 --arch=x86-64 -DEXPLICIT_VECTOR4=1
+// RUN: FileCheck %s --input-file=%t.asm --check-prefix=CHECK_VECTOR1
+
+// CHECK_VECTOR0:	[[LABEL:[a-zA-Z0-9_.]*]]:
+// CHECK_VECTOR0:		vmovups xmm0, xmmword ptr [rsi + 4*rcx]
```

Comment:
Not sure what is the intent here, but checking for specific code sequence, with specific registers is not a good idea most of the time, as there might be some legitimate variations.

In this specific issue (#2460), the good way to check the issue is gone, is to check for not generating "sign extends" (as `cdqe` and `movsxd`) instructios and check that loop counter increment is done for 64 bit values.

Reply:
WIll update the lit test to check that the sign extension is either not present or hoisted to the preheader of the loop by the induction simplification pass.